model/code2class.py

```diff
@@ -24,28 +25,43 @@ def __init__(
             len(self.vocab.type_to_id),
             self.vocab.type_to_id[PAD],
         )
-        self.classifier = PathClassifier(config.classifier_config, len(self.vocab.label_to_id))
+        self.num_classes = len(self.vocab.label_to_id)
+        self.classifier = PathClassifier(config.classifier_config, self.num_classes)
 
     def forward(self, samples: Dict[str, torch.Tensor], paths_for_label: List[int]) -> torch.Tensor:
         return self.classifier(self.encoder(samples), paths_for_label)
 
     def _general_epoch_end(self, outputs: List[Dict], loss_key: str, group: str) -> Dict:
         logs = {f"{group}/loss": torch.stack([out[loss_key] for out in outputs]).mean()}
-        progress_bar = {k: v for k, v in logs.items() if k in [f"{group}/loss"]}
+        accumulated_conf_matrix = torch.zeros(self.num_classes, self.num_classes, requires_grad=False)
+        for out in outputs:
+            _conf_matrix = out[f"{group}/confusion_matrix"]
+            max_class_index, _ = _conf_matrix.shape
+            accumulated_conf_matrix[:max_class_index, :max_class_index] += _conf_matrix
```

Comment:
If labels and predictions don't contain zero class, will the confusion matrix ignore it, or will there be row and column of zeros?